### PR TITLE
week5/assignment4: test cases using Jest

### PR DESCRIPTION
**Pull request by tusharwalzade216**
Create a simple Javascript function code for addition, subtraction, and multiplication of 2 numbers and write the corresponding Jest based tests for it.

```
const mathOperations = {
    sum: function(a, b) {
        return a + b;
    },
    diff: function(a, b) {
        return a - b;
    },
    product: function(a, b) {
        return a * b;
    }
}

module.exports = mathOperations;
```

## Guidelines:
1. Jest should've been installed.
2. Package.json file should have the config for running test cases.
3. Test case file should've been created. For example: calculator.test.js
4. BDD style tests for each function should've been created for the same.
5. All test cases should pass.
6. Reference: https://www.softwaretestinghelp.com/jest-testing-tutorial/

## Outcome:
1. Understanding the importance of writing test cases.
2. How BDD works.
3. What are the packages required for writing test cases?
4. How to configure test cases in package.json file.

### CODE REVIEW

File: Week-5/Assignment-4/src/calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const mathOperations = require("./calculator");
+
```
Comment (sairamin) [inline Week-5/Assignment-4/src/calculator.test.js:2]: Add more test cases

Comment (tusharwalzade216) [reply Week-5/Assignment-4/src/calculator.test.js:2]: Thanks @sairamin, added more test cases, please review and merge

File: Week-5/Assignment-4/src/calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+test("sum(2, 2) should be defined", () => {
+    expect(mathOperations.sum(2, 2)).toBeDefined();
+});
+
+test("sum(2, 2) should be 4", () => {
+    expect(mathOperations.sum(2, 2)).toBe(4);
+});
+
+test("sum(2, 2) should be greater than 3", () => {
```
Comment (sairamin) [inline Week-5/Assignment-4/src/calculator.test.js]: The intention of adding more test cases should reflect in choosing different set of arguments and not adding different assertion statements for same set of inputs.
What should happen let's say if i pass

1, undefined
1, null
undefined, undefined
1, "sds"
"sas", 1
"dsds", undefined

There are multiple scenarios like this. I expect you to cover those scenarios.

Comment (tusharwalzade216) [reply Week-5/Assignment-4/src/calculator.test.js]: Thanks @sairamin, I've updated my test cases now, please review

**Review by sairamin — APPROVED** [review]
Good job here @tusharwalzade216

**Pull request merged**